server-spi-private/src/main/java/org/keycloak/models/utils/RepresentationToModel.java

```diff
@@ -2817,12 +2817,7 @@ public static void createFederatedClientRoleMappings(UserFederatedStorageProvide
         }
 
         Map<String, String> m = new HashMap<>(map);
-        for (Iterator<Map.Entry<String, String>> itr = m.entrySet().iterator(); itr.hasNext(); ) {
-            Map.Entry<String, String> e = itr.next();
-            if (e.getValue() == null || e.getValue().equals("")) {
-                itr.remove();
-            }
-        }
+        m.entrySet().removeIf(e -> e.getValue() == null || e.getValue().equals(""));
```

Comment:
Ok then.

Reply:
@vmuzikar if I'm not mistaken, by default, `removeIf` uses an iterator internally to remove the elements so the performance should be exactly the same. I specifically checked this information in the docs during my review yesterday.

Reply:
Moreover, some sources say that `removeIf` is more optimized and faster as it first evaluates the predicate for all items before making any alterations to the list. [1]

[1] https://dzone.com/articles/the-way-of-the-lambda-and-removeif

Reply:
@Pepo48 I don't doubt it's much more elegant solution and maybe a little bit more optimized. :) But in general it seems to me like a bad smell that we make a change into a core code that is not actually related to the object of this PR: fixing a UI bug. I.e. it seems to me like out of scope for this PR, moreover when it can potentially affect other areas, not just IdPs.

Reply:
In other words, I'd rather see this change made (and discussed) in a separate PR. I'm quite sure we have the same issue at many more places, to it could be all address in another PR. ;)

Reply:
+1 :)